content/post/manage-redis-secret-engine-using-kubevault-in-aws/index.md

```diff
@@ -0,0 +1,587 @@
+kubevault   kubevault-kubevault-operator-86b8c7f688-6ln65         1/1     Running   0          22s
+kubevault   kubevault-kubevault-webhook-server-8554c7cd7f-j9w6q   1/1     Running   0          22s
+```
+#### Install KubeVault CLI
+
+KubeVault provides a `kubectl` plugin to interact with KubeVault resources. Let's install [KubeVault CLI](https://kubevault.com/docs/latest/setup/install/kubectl_plugin/)
+
+#### Install Secret-store CSI Driver
+
+```bash
+$ helm repo add secrets-store-csi-driver https://kubernetes-sigs.github.io/secrets-store-csi-driver/charts
+$ helm install csi-secrets-store secrets-store-csi-driver/secrets-store-csi-driver --namespace kube-system
+```
+#### Install Vault specific CSI Provider
+
+```bash
+$ helm repo add hashicorp https://helm.releases.hashicorp.com
+$ helm install vault hashicorp/vault \
+    --set "server.enabled=false" \
+    --set "injector.enabled=false" \
+    --set "csi.enabled=true"
+```
+
+### Create Namespace and Secret
+
+To keep everything isolated, we are going to use a separate namespace `demo` throughout this tutorial.
+
+```bash
+$ kubectl create namespace demo
+namespace/demo created
+```
+
+Here, We need to create a storage secret for our backend.
+
+```bash
+$ echo -n '<your-secret-key-id-here>' > secret_key
+$ echo -n '<your-secret-access-key-here>' > access_key
+$ kubectl create secret generic -n demo aws-secret \
+    --from-file=./secret_key \
+    --from-file=./access_key
+secret/aws-secret created
+```
+
+## Deploy VaultServer
+
+Now, we are going to deploy the `VaultServer`.
+
+```yaml
+apiVersion: kubevault.com/v1alpha2
+kind: VaultServer
+metadata:
+  name: vault
+  namespace: demo
+spec:
+  replicas: 1
+  version: 1.12.1
+  allowedSecretEngines:
+    namespaces:
+      from: All
+    secretEngines:
+      - redis
+  backend:
+    s3:
+      bucket: "vault-demo-1"
+      region: "us-east-1"
+      endpoint: s3.amazonaws.com
+      credentialSecretRef:
+        name: aws-secret
+  unsealer:
+    secretShares: 5
+    secretThreshold: 3
+    mode:
+      kubernetesSecret:
+        secretName: vault-keys
+  terminationPolicy: WipeOut
+```
+In this yaml, 
+- `spec.replicas` specifies the number of Vault nodes to deploy. It has to be a positive number.
+- `spec.version` specifies the name of the `VaultServerVersion` CRD. This CRD holds the image name and version of the Vault, Unsealer, and Exporter.
+- `spec.allowedSecretEngines` defines the Secret Engine informations which to be granted in this Vault Server.
+- `spec.backend` is a required field that contains the Vault backend storage configuration.
+- `spec.unsealer` specifies `Unsealer` configuration. `Unsealer` handles automatic initializing and unsealing of Vault.
+- `spec.terminationPolicy` is an optional field that gives flexibility whether to nullify(reject) the delete operation of `VaultServer` CRD or which resources KubeVault operator should keep or delete when you delete `VaultServer` CRD.
```

Comment:
`spec.terminationPolicy` field is *Wipeout* means that vault will be deleted without restrictions. It can also be "Halt", "Delete" and "DoNotTerminate". Learn More about these [HERE](https://kubevault.com/docs/v2022.12.28/concepts/vault-server-crds/vaultserver/#specterminationpolicy).